modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/endpoint/IndexEndpoint.java

```
@@ -152,6 +153,57 @@ public Response partiallyRebuildIndex(@PathParam("service") final String service
     return R.ok();
   }
 
+  @POST
+  @Path("rebuild/AssetManager/{part}")
+  @RestQuery(name = "partiallyRebuildIndexByPart",
```

I think I would like a more general endpoint here, even though right now the AssetManager is the only one who really supports this feature. Maybe we could have `rebuild/{service}/{part}` and then either

a) have the service itself throw an error if the part is not supported, letting the rebuild service handle that error and return a reasonable return code, or
b) add a getSupportedParts() method to the interface all services implement and have the rebuildservice check that before calling repopulate()

The result is the same, so I don't really care. But this will make this more extendable later and should hopefully not be that much more work.

---

While I generally agree, since this is not the main focus of this PR and I'm limited on time, I tried to not put too much effort into this while hopefully keeping it easily extendable for the future. But if you say your suggestion should really be part of this PR (for example to avoid changing the endpoint path down the line), I'll get to it.

---

I think it would be good to get this right from the get-go. Even though we don't guarantee anything for these endpoints, changing them sometime down the line is always a hassle.

---

Tried to rewrite the endpoint. Currently confused by https://github.com/opencast/opencast/blob/ccfe8dd415bbeea3b1f8a8ab1bda45b4ff2b54d2/modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/rebuild/IndexRebuildService.java#L212, which eats up exceptions during index rebuilds for reasons unknown to me. Will try to look further into this some other time. Furthermore, checking ServicePart in IndexEndpoint is not really possible anymore if ServicePart is moved to the actual service due to code architecture, so currently the endpoint always returns 200, no matter if the part was valid or not.

---

I just took a look at this. Swallowing the exception there is a bug imo, that was introduced by #4206. All the surrounding code expects that exception to bubble up. Just throwing that exception again after setting the rebuild state should solve your problem.

---

I can certainly throw an exception there, but this will not solve the problem that the index rebuild is running async (I think?), so I can't have it influence the return code.

---

Ah, of course, you're right about that. Sorry, that slipped my mind. Well, at least the UI will show you if it went wrong... IMO that exception should still be thrown, though. Adding to the list... I should stop looking at that code.